pkg/jx/cmd/get_applications.go

```diff
@@ -114,33 +116,116 @@ func NewCmdGetApplications(f Factory, in terminal.FileReader, out terminal.FileW
 
 // Run implements this command
 func (o *GetApplicationsOptions) Run() error {
-	f := o.Factory
-	client, currentNs, err := f.CreateJXClient()
+	kubeClient, _, err := o.KubeClient()
 	if err != nil {
 		return err
 	}
-	kubeClient, _, err := o.KubeClient()
+
+	namespaces, envApps, envNames, apps, err := o.getAppData(kubeClient)
 	if err != nil {
 		return err
 	}
+
+	util.ReverseStrings(namespaces)
+	if len(apps) == 0 {
+		log.Infof("No applications found in environments %s\n", strings.Join(envNames, ", "))
+		return nil
+	}
+	sort.Strings(apps)
+
+	o.Results.EnvApps = envApps
+	o.Results.EnvNames = envNames
+
+	table := o.generateTable(apps, envApps, kubeClient)
+
+	table.Render()
+	return nil
+}
+
+func (o *GetApplicationsOptions) generateTable(apps []string, envApps []EnvApps, kubeClient kubernetes.Interface) table.Table {
+	table := o.generateTableHeaders(envApps)
+
+	appEnvMap := map[string]map[string]*ApplicationEnvironmentInfo{}
+	for _, appName := range apps {
+		row := []string{appName}
+		for _, ea := range envApps {
+			version := ""
+			d := ea.Apps[appName]
+			appMap := appEnvMap[appName]
+			if appMap == nil {
+				appMap = map[string]*ApplicationEnvironmentInfo{}
+				appEnvMap[appName] = appMap
+			}
+			version = kube.GetVersion(&d.ObjectMeta)
+			appEnvInfo := &ApplicationEnvironmentInfo{
+				Deployment:  &d,
+				Environment: &ea.Environment,
+				Version:     version,
+			}
+			appMap[ea.Environment.Name] = appEnvInfo
+			if ea.Environment.Spec.Kind != v1.EnvironmentKindTypePreview {
+				row = append(row, version)
+			}
+			if !o.HidePod {
+				pods := ""
+				replicas := ""
+				ready := d.Status.ReadyReplicas
+				if d.Spec.Replicas != nil && ready > 0 {
+					replicas = formatInt32(*d.Spec.Replicas)
+					pods = formatInt32(ready) + "/" + replicas
+				}
+				row = append(row, pods)
+			}
+			if !o.HideUrl {
+				url, _ := services.FindServiceURL(kubeClient, d.Namespace, appName)
+				if url == "" {
+					url, _ = services.FindServiceURL(kubeClient, d.Namespace, d.Name)
+				}
+				if url == "" {
+					// handle helm3
+					chart := d.Labels["chart"]
```

Comment:
check if the deployment really has a `chart` label: e.g. `chart, ok := d.Labels["chart"]`